src/makeCalculator.js

```diff
@@ -4,7 +4,47 @@
  * @return {object}
  */
 function makeCalculator() {
-  // write code here
+  // let result = 0;
+
+  return {
+    result: 0,
+
+    add(value) {
+      this.result += value;
+
+      return this.result;
+    },
+
+    subtract(value) {
+      this.result -= value;
+
+      return this.result;
+    },
+
+    multiply(value) {
+      this.result *= value;
+
+      return this.result;
+    },
+
+    divide(value) {
+      this.result /= value;
+
+      return this.result;
+    },
+
+    reset() {
+      this.result = 0;
+
+      return this;
+    },
+
+    operate(callback, value) {
+      this.result = callback.call(this, value);
```

Comment:
```suggestion
      callback.call(this, value);
```
looks like redundant to reassign this.result.
And in general no need to return this.result from each operate method. enough just calculate this.result.